src/main/kotlin/app/UserService.kt

```diff
@@ -25,45 +22,33 @@ object UserService {
-        val remainingRequests by lazy { GhService.remainingRequests }
-        val hasFreeRemainingRequests by lazy { remainingRequests > (freeRequestCutoff ?: remainingRequests) }
+        val userCacheJson by lazy { CacheService.selectJsonFromDb(user) }
         return Config.unrestricted()
-                || Cache.contains(user)
-                || hasFreeRemainingRequests
-                || (remainingRequests > 0 && hasStarredRepo(user))
+            || (userCacheJson != null)
+            || lazy { hasFreeRemainingRequests() }.value
+            || (lazy { remainingRequests() }.value > 0 && hasStarredRepo(user))
     }
 
-    fun getUserProfile(username: String): UserProfile {
-        if (Cache.invalid(username)) {
-            val user = GhService.users.getUser(username)
-            val repos = GhService.repos.getRepositories(username).filter { !it.isFork && it.size != 0 }
-            val repoCommits = repos.parallelStream().map { it to commitsForRepo(it).filter { it.author?.login.equals(username, ignoreCase = true) } }.toList().toMap()
-            val langRepoGrouping = repos.groupingBy { (it.language ?: "Unknown") }
-
-            val quarterCommitCount = CommitCountUtil.getCommitsForQuarters(user, repoCommits)
-            val langRepoCount = langRepoGrouping.eachCount().toList().sortedBy { (_, v) -> -v }.toMap()
-            val langStarCount = langRepoGrouping.fold(0) { acc, repo -> acc + repo.watchers }.toList().sortedBy { (_, v) -> -v }.toMap()
-            val langCommitCount = langRepoGrouping.fold(0) { acc, repo -> acc + repoCommits[repo]!!.size }.toList().sortedBy { (_, v) -> -v }.toMap()
-            val repoCommitCount = repoCommits.map { it.key.name to it.value.size }.toList().sortedBy { (_, v) -> -v }.take(10).toMap()
-            val repoStarCount = repos.filter { it.watchers > 0 }.map { it.name to it.watchers }.sortedBy { (_, v) -> -v }.take(10).toMap()
-
-            val repoCommitCountDescriptions = repoCommitCount.map { it.key to repos.find { r -> r.name == it.key }?.description }.toMap()
-            val repoStarCountDescriptions = repoStarCount.map { it.key to repos.find { r -> r.name == it.key }?.description }.toMap()
-
-            Cache.putUserProfile(UserProfile(
-                    user,
-                    quarterCommitCount,
-                    langRepoCount,
-                    langStarCount,
-                    langCommitCount,
-                    repoCommitCount,
-                    repoStarCount,
-                    repoCommitCountDescriptions,
-                    repoStarCountDescriptions
-            ))
+    fun getUserIfCanLoad(username: String): UserProfile {
+        val userCacheJson by lazy { CacheService.selectJsonFromDb(username) }
+        val canLoadUser = Config.unrestricted()
+            || (userCacheJson != null)
+            || lazy { hasFreeRemainingRequests() }.value
+            || (lazy { remainingRequests() }.value > 0 && hasStarredRepo(username))
```

Comment:
can also remove lazy here

Reply:
[33d6994](https://github.com/tipsy/profile-summary-for-github/pull/113/commits/33d69948f54ccb5403500484a1eecefc569bc2a1)